Upgrade to Phoenix 1.6.16

Emina-M opened the pull request:
- Upgrade Phoenix scaffolding from version `1.5.3` to `1.6.16` using generator:

```
mix phx.new astarte_flow --app astarte_flow --module Astarte.Flow \
--no-assets --no-ecto --no-html --no-live --no-mailer --no-dashboard
```

- Fix package tarball prefix in `.gitignore`
- Update dependencies

Emina-M committed:
Upgrade to Phoenix 1.6.16

Upgrade Phoenix scaffolding from version 1.5.3 to 1.6.16 using
generator:

mix phx.new astarte_flow --app astarte_flow --module Astarte.Flow \
--no-assets --no-ecto --no-html --no-live --no-mailer --no-dashboard

Signed-off-by: Emina Muminovic <emina.muminovic@secomind.com>

Emina-M committed:
Fix package tarball prefix in .gitignore

Replace `astarte_streams` with `astarte_flow`.

Signed-off-by: Emina Muminovic <emina.muminovic@secomind.com>

Annopaolo commented on lib/astarte_flow.ex:
No need for this file

Annopaolo commented on lib/astarte_flow_web.ex:
```diff
@@ -52,17 +52,18 @@ defmodule Astarte.FlowWeb do
 
-      import Astarte.FlowWeb.ErrorHelpers
-      import Astarte.FlowWeb.Gettext
-      alias Astarte.FlowWeb.Router.Helpers, as: Routes
+      # Include shared imports and aliases for views
+      unquote(view_helpers())
```
Why the need for quoting/unquoting imports and aliases here?

Emina-M replied:
Most of the changes in this PR follow scaffolding generated with phoenix 1.6.16 generator. The goal is to be consistent with generated code, so future upgrades are easier. If it's not something we adhere to, I can remove this change.

Annopaolo commented on mix.exs:
```diff
@@ -27,7 +27,7 @@ defmodule Astarte.Flow.MixProject do
       app: :astarte_flow,
       version: "0.1.0",
-      elixir: "~> 1.7",
+      elixir: "~> 1.12",
```
Shouldn't this be  `1.17` now?

Emina-M replied:
Similar answer to previous comment. You can take a look at the versions diff [here](https://www.phoenixdiff.org/compare/1.5.3%20--live...1.6.16)

Emina-M committed:
Update dependencies

Autogenerated with `mix deps.update --all`

Major changes:
telemetry_metrics 0.6 => 1.1.0
telemetry_metrics_prometheus_core 0.4 => 1.2.1

Signed-off-by: Emina Muminovic <emina.muminovic@secomind.com>
